### DISCUSSION TITLE
数据更新时的异常

### DISCUSSION THREAD

**Discussion by zwac00001** [zuihou/zuihou-admin-cloud#287] [01.常见问题]
1.当这条数据是从库里面查询出来的，修改某个值后upadte，修改时间并未发生变动；
2.对某个值置空，修改时空值并未参与修改

**Comment by zuihou** [comment]
1. 原理参考： LampMetaObjectHandler
```
if (metaObject.hasGetter(Entity.UPDATED_TIME)) {
    Object oldVal = metaObject.getValue(Entity.UPDATED_TIME);

     // 修改时间 == null 时，才会自动赋值。
    if (oldVal == null) {
        this.setFieldValByName(Entity.UPDATED_TIME, LocalDateTime.now(), metaObject);
    }
}
```
解决方案：
1. 从库里面查询出来后，手动置空
2. 修改LampMetaObjectHandler源码

**Comment by zuihou** [comment]
2. mybaits-plus 设计如此，请认真查阅mp官方文档，寻找解决方案。
- updateStrategy ：   https://www.baomidou.com/pages/56bac0/#updatestrategy
- fieldstrategy： https://www.baomidou.com/pages/223848/#fieldstrategy
- TableField ：  https://www.baomidou.com/pages/223848/#tablefield
- issue：  https://gitee.com/baomidou/mybatis-plus/issues?utf8=%E2%9C%93&state=all&issue_search=updateById
-
